test/e2e/apps/livenessprobemapnodeprobe.go

```diff
@@ -0,0 +1,291 @@
+			containerLivenessProbeConfig := v1.Probe{
+				ProbeHandler: v1.ProbeHandler{
+					Exec: &v1.ExecAction{
+						Command: []string{"/bin/sh", "-c", "/healthy.sh"},
+					},
+				},
+				InitialDelaySeconds: 1000,
+				TimeoutSeconds:      5,
+				PeriodSeconds:       100,
+				SuccessThreshold:    1,
+				FailureThreshold:    3,
+			}
+			cName := "app"
+			containersProbe := []livenessprobeUtils.ContainerLivenessProbe{
+				{
+					Name:          cName,
+					LivenessProbe: containerLivenessProbeConfig,
+				},
+			}
+			expectContainerProbeConfigRaw, err := json.Marshal(containersProbe)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			ginkgo.By("Create CloneSet and wait 1 pod ready")
+			pods = tester.CreateTestCloneSetAndGetPods(randStr, 1, map[string]string{
+				alpha1.AnnotationUsingEnhancedLiveness: "true",
+			}, nil, []v1.Container{
+				{
+					Name:          cName,
+					Image:         WebserverImage,
+					LivenessProbe: &containerLivenessProbeConfig,
+				},
+			})
+			ginkgo.By(fmt.Sprintf("Assert pod annotation %v", alpha1.AnnotationNativeContainerProbeContext))
+			pod := pods[0]
+			gomega.Eventually(func() string {
+				return pod.Annotations[alpha1.AnnotationNativeContainerProbeContext]
+			}, 5*time.Second, 1*time.Second).Should(gomega.Equal(string(expectContainerProbeConfigRaw)))
+
+			ginkgo.By("Wait NodePodProbe create completion")
+			gomega.Eventually(func() *appsv1alpha1.NodePodProbe {
+				gotNodePodProbe, err := tester.GetNodePodProbe(pod.Spec.NodeName)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				return gotNodePodProbe
+			}, 70*time.Second, time.Second).ShouldNot(gomega.BeNil())
+
+			ginkgo.By("Check the generated nodePodProbe config")
+			expectNodePodProbeSpec := []appsv1alpha1.PodProbe{
+				{
+					Name:      pod.Name,
+					Namespace: pod.Namespace,
+					Probes: []appsv1alpha1.ContainerProbe{
+						{
+							ContainerName: cName,
+							Name:          fmt.Sprintf("%v-%v", pod.Name, cName),
+							Probe: appsv1alpha1.ContainerProbeSpec{
+								Probe: v1.Probe{
+									ProbeHandler: v1.ProbeHandler{
+										Exec: &v1.ExecAction{
+											Command: []string{"/bin/sh", "-c", "/healthy.sh"},
+										},
+									},
+									InitialDelaySeconds: 1000,
+									TimeoutSeconds:      5,
+									PeriodSeconds:       100,
+									SuccessThreshold:    1,
+									FailureThreshold:    3,
+								},
+							},
+						},
+					},
+					UID: fmt.Sprintf("%v", pod.UID),
+				},
+			}
+			gomega.Eventually(func() string {
+				gotNodePodProbe, err := tester.GetNodePodProbe(pod.Spec.NodeName)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				podProbe := gotNodePodProbe.Spec.PodProbes
+				return util.DumpJSON(podProbe)
+			}, 70*time.Second, time.Second).Should(gomega.Equal(util.DumpJSON(expectNodePodProbeSpec)))
+		})
+
+		framework.ConformanceIt("case:2 Create one pod with liveness probe config and update the", func() {
```

Comment:
case:2 Create one pod with liveness probe config and update the -> case:2 Create one pod with liveness probe config and update the nodePodProbe

Reply:
done